service/src/main/java/com/epam/esm/service/service/tag/TagServiceImpl.java

```diff
@@ -184,4 +189,25 @@ public Tag readMostlyUsedTag() throws NotFoundException {
             return tag;
         }
     }
+
+    @Override
+    public Tag getByUniqueName(String name) throws NotFoundException {
+        Optional<Tag> tagOptional = getByUniqueNameOptional(name);
+        if (tagOptional.isEmpty()) {
+            throw new NotFoundException("Requested resource not found(name = " + name + ")!",
+                    ErrorCodes.TAG_NOT_FOUND);
+        }
+        return tagOptional.get();
+    }
+
+    @Override
+    public Optional<Tag> getByUniqueNameOptional(String name) {
+        TagFinder finder = getFinder();
+        addToFinder(finder::findByName, name);
+        List<Tag> tags = dao.findByParameters(finder);
+        if (CollectionUtils.isEmpty(tags)) {
+            return Optional.empty();
+        }
+        return Optional.of(tags.get(0));
```

Comment:
unify if block and return by the ternary operator

Reply:
Done